src/ripple/basics/Buffer.h

```diff
@@ -58,33 +93,28 @@ class Buffer
     */
     Buffer& operator= (Buffer&& other)
     {
-        p_ = std::move(other.p_);
-        size_ = other.size_;
-        other.size_ = 0;
+        if (this != &other)
+        {
+            p_ = std::move(other.p_);
+            size_ = other.size_;
+            other.size_ = 0;
+        }
         return *this;
     }
 
-    /** Create an uninitialized buffer with the given size. */
+    /** Construct from a slice */
     explicit
-    Buffer (std::size_t size)
-        : p_ (size ?
-            new std::uint8_t[size] : nullptr)
-        , size_ (size)
-    {
-    }
-
-    /** Create a buffer as a copy of existing memory. */
-    Buffer (void const* data, std::size_t size)
-        : Buffer (size)
+    Buffer (Slice s)
+        : Buffer (s.data(), s.size())
     {
-        std::memcpy(p_.get(), data, size);
     }
 
-    /** Create a buffer from a copy of existing memory. */
-    explicit
-    Buffer (Slice const& slice)
-        : Buffer(slice.data(), slice.size())
+    /** Assign from slice */
+    Buffer& operator= (Slice s)
```

Comment:
Great question.  Short answer: yes, I think it's worth a check.  I spent some time thinking about this question.  Consider:
```
std::string test { "Hello World" };

Buffer b (test.data(), test.size());
Slice s (b.data(), b.size());
s += 1;
b = s; // copies from released memory.
```
It looks silly when it's all in line like this, but spread across two or more functions this would be an easy mistake to make.

I think the problem is fixable by noting whether `s.data()` sits within the range `&p_[0]` to ` &p_[size_-1]`.  The complexity doesn't need to leak out of this particular assignment operator.  So, yeah, I think it's worth addressing.  Another viable choice would be to remove this assignment operator and only allow construction (not assignment) from a `Slice`.